query/src/provider.rs

```diff
@@ -3167,6 +3167,201 @@ mod test {
+        ) as Arc<dyn QueryChunk>;
+
+        // chunk2_6 no overlap, duplicates within
+        let chunk2_6 = Arc::new(
+            TestChunk::new("t")
+                .with_id(4)
+                .with_time_column()
+                .with_timestamp_min_max(38000, 40000)
+                .with_tag_column("tag1")
+                .with_i64_field_column("field_int")
+                .with_partition_id(2) // signal the chunk in partition 1 while grouping overlaps
+                .with_may_contain_pk_duplicates(true) // signal having duplicates within this chunk
+                .with_sort_key(sort_key.clone()), // signal the chunk is sorted
+        ) as Arc<dyn QueryChunk>;
+
+        let schema = chunk1_1.schema();
+        let chunks = vec![
+            chunk1_1, chunk1_2, chunk1_3, chunk1_4, chunk2_1, chunk2_2, chunk2_3, chunk2_4,
+            chunk2_5, chunk2_6,
+        ];
+        let mut deduplicator = Deduplicater::new();
+        let plan = deduplicator
+            .build_scan_plan(
+                Arc::from("t"),
+                schema,
+                chunks,
+                Predicate::default(),
+                Some(sort_key),
+            )
+            .unwrap();
+
+        // The plan should look like this. No SortExec at all because
+        // all chunks are already sorted on the same requested sort key
+        //
+        // SortPreservingMergeExec: [tag1@1 ASC,time@2 ASC]
+        //     UnionExec
+        //     DeduplicateExec: [tag1@1 ASC,time@2 ASC]
+        //         SortPreservingMergeExec: [tag1@1 ASC,time@2 ASC]
+        //             UnionExec
+        //                 IOxReadFilterNode: table_name=t, chunks=1 predicate=Predicate
+        //                 IOxReadFilterNode: table_name=t, chunks=1 predicate=Predicate
+        //     DeduplicateExec: [tag1@1 ASC,time@2 ASC]
+        //         SortPreservingMergeExec: [tag1@1 ASC,time@2 ASC]
+        //             UnionExec
+        //                 IOxReadFilterNode: table_name=t, chunks=1 predicate=Predicate
+        //                 IOxReadFilterNode: table_name=t, chunks=1 predicate=Predicate
+        //     DeduplicateExec: [tag1@1 ASC,time@2 ASC]
+        //         IOxReadFilterNode: table_name=t, chunks=1 predicate=Predicate
+        //     DeduplicateExec: [tag1@1 ASC,time@2 ASC]
+        //         IOxReadFilterNode: table_name=t, chunks=1 predicate=Predicate
+        //     DeduplicateExec: [tag1@1 ASC,time@2 ASC]
+        //         IOxReadFilterNode: table_name=t, chunks=1 predicate=Predicate
+        //     IOxReadFilterNode: table_name=t, chunks=1 predicate=Predicate
+        //     IOxReadFilterNode: table_name=t, chunks=1 predicate=Predicate
+        //     IOxReadFilterNode: table_name=t, chunks=1 predicate=Predicate
+        let plan_str = format!("{}", displayable(plan.as_ref()).indent());
+        //println!("{}", plan_str);
+
+        //  Verify no SortExec because chunks are not yet sorted on the specified sort_key
```

Comment:
This comment seems to conflict with the comment above that says:

```
        // The plan should look like this. No SortExec at all because
        // all chunks are already sorted on the same requested sort key
```

So I think this comment should say: 

```suggestion
        //  Verify no SortExec because chunks are already sorted on the specified sort_key
```

?